internal/directives/file_deleter_test.go

```diff
@@ -0,0 +1,120 @@
+package directives
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+)
+
+func Test_fileDeleter_runPromotionStep(t *testing.T) {
+	tests := []struct {
+		name       string
+		setupFiles func(*testing.T) string
+		cfg        DeleteConfig
+		assertions func(*testing.T, string, PromotionStepResult, error)
+	}{
+		{
+			name: "succeeds deleting file",
+			setupFiles: func(t *testing.T) string {
+				tmpDir := t.TempDir()
+
+				path := filepath.Join(tmpDir, "input.txt")
+				require.NoError(t, os.WriteFile(path, []byte("test content"), 0o600))
+
+				return tmpDir
+			},
+			cfg: DeleteConfig{
+				Path: "input.txt",
+			},
+			assertions: func(t *testing.T, _ string, result PromotionStepResult, err error) {
+				assert.NoError(t, err)
+				assert.Equal(t, PromotionStepResult{Status: kargoapi.PromotionPhaseSucceeded}, result)
+
+				_, statError := os.Stat("input.txt")
+				assert.ErrorIs(t, statError, os.ErrNotExist)
+			},
+		},
+		{
+			name: "succeeds deleting directory",
+			setupFiles: func(t *testing.T) string {
+				tmpDir := t.TempDir()
+				dirPath := filepath.Join(tmpDir, "dirToDelete")
+				require.NoError(t, os.Mkdir(dirPath, 0o700))
+				return tmpDir
+			},
+			cfg: DeleteConfig{
+				Path: "dirToDelete",
+			},
+			assertions: func(t *testing.T, workDir string, result PromotionStepResult, err error) {
+				assert.NoError(t, err)
+				assert.Equal(t, PromotionStepResult{Status: kargoapi.PromotionPhaseSucceeded}, result)
+
+				_, statErr := os.Stat(filepath.Join(workDir, "dirToDelete"))
+				assert.ErrorIs(t, statErr, os.ErrNotExist)
+			},
+		},
+		{
+			name: "fails for non-existent path",
+			setupFiles: func(t *testing.T) string {
+				return t.TempDir()
+			},
+			cfg: DeleteConfig{
+				Path: "nonExistentFile.txt",
+			},
+			assertions: func(t *testing.T, _ string, result PromotionStepResult, err error) {
+				assert.Error(t, err)
+				assert.Equal(t, PromotionStepResult{Status: kargoapi.PromotionPhaseErrored}, result)
+			},
+		},
+		{
+			name: "removes symlink only",
```

Comment:
```suggestion
			name: "removes symlink only",
```

@hiddeco I tried debugging to understand why is this test case failing even though the path specified is correct. After debugging i found that, `SecureJoin()` is actually joining the file `input/input.txt` that is linked by the `symlink.txt` instead of joining the file `symlink.txt`, Following is the error message

```
=== RUN   Test_fileDeleter_runPromotionStep
--- FAIL: Test_fileDeleter_runPromotionStep (120.72s)
=== RUN   Test_fileDeleter_runPromotionStep/removes_symlink_only
path to delete: /tmp/Test_fileDeleter_runPromotionStepremoves_symlink_only1504252788/001/input/input.txt
    file_deleter_test.go:100: 
        	Error Trace:	/home/nitish/Documents/Work/Open-Source/kargo/internal/directives/file_deleter_test.go:100
        	            				/home/nitish/Documents/Work/Open-Source/kargo/internal/directives/file_deleter_test.go:115
        	Error:      	Received unexpected error:
        	            	stat /tmp/Test_fileDeleter_runPromotionStepremoves_symlink_only1504252788/001/input.txt: no such file or directory
        	Test:       	Test_fileDeleter_runPromotionStep/removes_symlink_only
    --- FAIL: Test_fileDeleter_runPromotionStep/removes_symlink_only (120.72s)
```